python/cog/types.py

```diff
@@ -242,19 +237,25 @@ def __delattr__(self, name: str) -> None:
 
     # Luckily the only dunder method on HTTPResponse is __iter__
     def __iter__(self) -> Iterator[bytes]:
-        return iter(self.__wrapped__)
+        response = self.__wrapped__
+        return iter(response)
 
     @property
     def __wrapped__(self) -> Any:
         try:
             return object.__getattribute__(self, "__target__")
         except AttributeError:
             url = object.__getattribute__(self, "__url__")
-            resp = requests.get(url, stream=True)
-            resp.raise_for_status()
-            resp.raw.decode_content = True
-            object.__setattr__(self, "__target__", resp.raw)
-            return resp.raw
+
+            # We create a streaming response here, much like the `requests`
+            # version in the main 0.9.x branch. The only concerning bit here
+            # is that the book keeping for closing the response needs to be
+            # handled elsewhere. There's probably a better design for this
+            # in the long term.
+            res = urllib.request.urlopen(url)  # noqa: S310
```

Comment:
I think we need to pre-validate that this is an `http` or `https` URL. (Do we also support `data` URLs?)

Either way, we shouldn't allow this:

```
In [1]: len(urllib.request.urlopen("file:///etc/shadow").read())
Out[1]: 9050
```